frontend/src/lib/candidate/components/PasswordSetComponent/PasswordSetComponent.svelte

```diff
@@ -0,0 +1,77 @@
+<script lang="ts">
+  import {t} from '$lib/i18n';
+  import {Button} from '$lib/components/button';
+  import PasswordField from '../PasswordField/PasswordField.svelte';
+  import {getRoute, Route} from '$lib/utils/navigation';
+  import PasswordValidator from '../passwordValidator/PasswordValidator.svelte';
+  import {createEventDispatcher} from 'svelte';
+  export let password1 = '';
+  export let password2 = '';
+  export let autocomplete1 = '';
+  export let autocomplete2 = '';
+  export let errorMessage = '';
+  export let validPassword = false;
+
+  $: disableSetButton = validPassword && password2.length > 0;
+  //dispatcher used for a function that calls the submit button's function in parent component
+  const dispatch = createEventDispatcher();
+  export function ButtonPressed() {
+    dispatch('ButtonPressed');
+  }
```

Comment:
Why is an event dispatched here? I think instead of dispatching an event and catching it, it would be more sensible to provide a callback function as a prop and assigning it to the submit event.

Reply:
Also make sure that the variables follow our naming conventions. The `ButtonPressed` function should be in camel case.